libs/langchain/langchain/tools/youtube/search.py

```diff
@@ -32,8 +32,8 @@ def _search(self, person: str, num_results: int) -> str:
 
         results = YoutubeSearch(person, num_results).to_json()
         data = json.loads(results)
-        url_suffix_list = [video["url_suffix"] for video in data["videos"]]
-        return str(url_suffix_list)
+        url_suffix_list = ["https://www.youtube.com" + video["url_suffix"] for video in data["videos"]]
+        return url_suffix_list
```

Comment:
1) Is the base URL available in the raw results 
https://github.com/joetats/youtube_search/blob/master/youtube_search/__init__.py#L39

2) _search returns a string as it's an input into an LLM, so we need to keep a cast to `str`

Reply:
1. No, the base URL is not available in the raw results; we only have the URL suffix starting with `"/watch?v="` in the raw data. You can refer to the JSON file returned [here](https://github.com/ansilmbabl/dump/blob/fa83f703428969637835bb54159c7a1088af714b/Open-source%20/Langchain/YouTubeSearchTool/video_data.json#L69) for a detailed view of the data structure.

2. Why we need to return the YouTube links to an LLM when it can't analyze the content within those links? Our primary goal is to obtain the video links themselves. Please correct me if my understanding is incorrect :)

Reply:
@eyurtsev tools can actually return `Any` - @ansilmbabl if you want to add back in that change in a future PR we can probably accept it

Reply:
thank you, I will create a PR then